tools/dds/dds-sniffer/dds-sniffer.cpp

```diff
@@ -0,0 +1,280 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include "dds-sniffer.hpp"
+
+#include <thread>
+
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/domain/qos/DomainParticipantQos.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/dds/subscriber/SampleInfo.hpp>
+#include <fastrtps/types/DynamicDataHelper.hpp>
+#include <fastrtps/types/DynamicDataFactory.h>
+
+#include <fastrtps/transport/UDPv4TransportDescriptor.h>
+
+
+int main(int argc, char** argv)
+{
+    eprosima::fastdds::dds::DomainId_t domain = 0;
+    uint32_t seconds = 0;
+
+    for(int i = 2; i <= argc; ++i)
+    {
+        if (strcmp(argv[i - 1], "--help") == 0 ||
```

Comment:
Consider using `tclap` 3-rd party for handing application argument.
See here: https://github.com/Nir-Az/librealsense/blob/9895ddfd2efc1d6752b5d1701f38c9fc6e234054/tools/fw-update/rs-fw-update.cpp#L139